### PR TITLE
Add storage behaviour

### PR DESCRIPTION
**Pull request by kpanic**

### CODE REVIEW

File: apps/exalice/lib/exalice/storage.ex [hunk 1]
```diff
@@ -0,0 +1,7 @@
+defmodule Storage do
+
+  @callback geocode(module :: atom, module :: atom, address :: String.t) :: Map.t
```
Comment (mouadino) [inline apps/exalice/lib/exalice/storage.ex]: duplicated "module", is that intentional?

Comment (kpanic) [inline apps/exalice/lib/exalice/storage.ex]: It was intentional, however I will rename for the sake of clarity, thank you! :)

File: apps/exalice/lib/exalice/geocoder.ex [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-      {:ok, 200, _} = Elastic.Indexer.index([location])
+  use HTTPoison.Base
 
+  def geocode(storage \\ ExAlice.Geocoder.config(:provider),
+              geocoder \\ ExAlice.Geocoder.config(:geocoder),
+              where) do
+    address = storage.geocode(where)
+    if Enum.empty?(address) do
+      {:ok, address} = geocoder.geocode(where)
+      {:ok, 200, _} = store(storage, [address])
```
Comment (benwilson512) [inline apps/exalice/lib/exalice/geocoder.ex:12]: You may want to consider using `with` here to return errors.

Comment (kpanic) [inline apps/exalice/lib/exalice/geocoder.ex:12]: thanks @benwilson512 ! I will merge this and address it later on

**Pull request merged**